solutions/usaco-894.mdx

```diff
@@ -90,6 +127,61 @@ valid assignment of grass types in the $\texttt{ans}$ array.
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+// USACO-style file input
+void setIO(string name = ""){ // FastIO see General -> Fast Input and Output
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	if(name.size()){
+		freopen((name+".in").c_str(), "r", stdin);
+		freopen((name+".out").c_str(), "w", stdout);
+	}
+}
+
+bool visited[10001];
+int ans[10001];
+vector<int> adjList[10001];
```

Comment:
magic numbers are kinda bad
perhaps just have a `MAX_N`, set it to 10k, and have these be size `MAX_N + 1`